docs/library/index.rst

```diff
@@ -1,21 +1,239 @@
-Libraries
-===============================================================================
+.. _axi_pwm_gen:
+
+AXI PWM Generator
+================================================================================
+
+.. hdl-component-diagram::
+
+The :git-hdl:`AXI PWM Generator <library/axi_pwm_gen>` core is used to generate
+a maximum of 16 configurable signals (Pulse-Width Modulations).
+The pulses are generated according to the state of a counter; there is one counter
+for each pulse.
+
+Features
+--------------------------------------------------------------------------------
+
+* Up to 16 configurable signals(period, width, offset)
+* External synchronization
+* External clock
+
+
+To disable a PWM channel, write 0 to its ``period`` register.
+
+The offset feature can synchronize channels 0 to 15 relative to an offset counter.
+
+The below features can be enabled by setting a flag ar register REG_UP_CONTROL(0x18)
+
+1. Channel phase alignment
+The AXI PWM Generator core can be synchronized by an external signal on the
+pulse HIGH -> LOW transition.
+The external sync ca be used in two modes, based on the external sync align feature.
+If external_sync_align flag is active the ext_sync will trigger a phase align
+at each neg-edge. Otherwise the phase align must be armed by a load config toggle,
+while the external sync must be heal HIGH.
+
+2. Software bringup(software reset)
+If active the software must bring the core out
+of reset, after a system reset, in order for the pwm signals to be generated.
+
+3. Force align.
+If active the current active pulses are immediately stopped and
+realigned. Otherwise, the synchronized pulses will start only after all running
+pulse periods end. Software over writable at runtime.
+
+4. Start at sync.
+If active, the pulses will start after the trigger event.
+Otherwise each pulse will start after a period equal to the one for which
+it is set. Software over writable at runtime.
+
+Default value given at build time. Software over-writable at runtime.
+- software bringup
+- start at sync
+- force align
+- ext sync align
```

Comment:
This is not rendered as a list because there needs to be a space before "-" as indentation.
And preferably, an empty line after the list.

```suggestion
 - software bringup
 - start at sync
 - force align
 - ext sync align
 
```